src/args.c

```diff
@@ -123,10 +123,10 @@ int ashuffle_options(struct ashuffle_options * opts,
             match_field = NULL;
             state = RULE;
         } else if (state == QUEUE) {
-            opts->queue_only = (unsigned) strtoul(argv[i], NULL, 10);
-            /* Make sure we got a valid queue number */
-            if (errno == EINVAL || errno == ERANGE) {
-                fputs("Error converting queue length to integer.\n", stderr);
+            opts->queue_only = (unsigned) strtoul(argv[i], &match_field, 10);
```

Comment:
Using `match_field` for this check is a little confusing. We should use a different (probably local) variable for it. Or, since this check is also duplicated below, just make a function that performs this check.

Reply:
Yeah, I did think reusing match_field was a bit overdoing it, that's what I get for being lazy :p
Making it into a helper function would be great but I'm not quite sure how you'd want this to work because the return space is limited (we're already fitting an unsigned long into an unsigned int without checking for overflow.. hmm actually I guess that'd be a good timing to add that as well probably)
function returns UINT_MAX and sets errno on error?